docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/Text/ProduceNgrams.cs

```diff
@@ -0,0 +1,88 @@
+            var mlContext = new MLContext();
+
+            // Create a small dataset as an IEnumerable.
+            var samples = new List<TextData>()
+            {
+                new TextData(){ Text = "This is an example to compute Ngrams." },
+                new TextData(){ Text = "Ngram is a sequence of 'N' consecutive words/tokens." },
+                new TextData(){ Text = "ML.NET's ProduceNgrams API produces vector of Ngrams." },
+                new TextData(){ Text = "Each position in the vector corresponds to a particular Ngram." },
+                new TextData(){ Text = "The value at each position corresponds to," },
+                new TextData(){ Text = "the number of times Ngram occured in the data (Tf), or" },
+                new TextData(){ Text = "the inverse of the number of documents that contain the Ngram (Idf), or." },
+                new TextData(){ Text = "or compute both and multipy together (Tf-Idf)." },
+            };
+
+            // Convert training data to IDataView.
+            var dataview = mlContext.Data.LoadFromEnumerable(samples);
+
+            // A pipeline for converting text into numeric Ngram features.
+            // The following call to 'ProduceNgrams' requires the tokenized text/string as input.
+            // This is acheived by calling 'TokenizeIntoWords' first followed by 'ProduceNgrams'.
+            // Please note that the length of the output feature vector depends on the Ngram settings.
+            var textPipeline = mlContext.Transforms.Text.TokenizeIntoWords("Tokens", "Text")
+                .Append(mlContext.Transforms.Conversion.MapValueToKey("Tokens"))
+                .Append(mlContext.Transforms.Text.ProduceNgrams("NgramFeatures", "Tokens",
+                ngramLength: 3, useAllLengths: false, weighting: NgramExtractingEstimator.WeightingCriteria.Tf));
+            
+            // Fit to data.
+            var textTransformer = textPipeline.Fit(dataview);
+            var transformedDataView = textTransformer.Transform(dataview);
+
+            // Create the prediction engine to get the Ngram features extracted from the text.
+            var predictionEngine = mlContext.Model.CreatePredictionEngine<TextData, TransformedTextData>(textTransformer);
+
+            // Convert the text into numeric features.
+            var prediction = predictionEngine.Predict(samples[0]);
+
+            // Print the length of the feature vector.
+            Console.WriteLine($"Number of Features: {prediction.NgramFeatures.Length}");
+
+            // Preview of the produced .
+            VBuffer<ReadOnlyMemory<char>> slotNames = default;
+            transformedDataView.Schema["NgramFeatures"].GetSlotNames(ref slotNames);
+            var NgramFeaturesColumn = transformedDataView.GetColumn<VBuffer<float>>(transformedDataView.Schema["NgramFeatures"]);
+            var slots = slotNames.GetValues();
+            Console.Write("Ngrams: ");
```

Comment:
>  Console.Write("Ngrams: "); [](start = 10, length = 28)

i'd remove this too.

Reply:
Actually, I'd like to keep it because its more clear to read each line with this prefix on console.

---
In reply to: [271816588](https://github.com/dotnet/machinelearning/pull/3177#discussion_r271816588) [](ancestors = 271816588)